test_schedule.py

```diff
@@ -120,6 +120,10 @@ def test_next_run_time(self):
             assert every().friday.do(mock_job).next_run.day == 8
             assert every().saturday.do(mock_job).next_run.day == 9
             assert every().sunday.do(mock_job).next_run.day == 10
+            assert every().minute.until('12:17').do(
+                mock_job).next_run.minute == 16
+            assert every().minute.until('2010-01-06 09:00').do(
+                mock_job).next_run.year == 9999
```

Comment:
I think we can add some more test cases here. For example: what would happen if someone enters an invalid date, for example `33:77`?

And we need to add tests to validate that a job is actually canceled after the deadline is reached.